Marketplace/giphy-bubble/app/giphyMoodApp.js

```diff
@@ -0,0 +1,206 @@
+            Entities.editEntity(giphyMoodWebEntityID, { sourceUrl : newURL});
+            tablet.emitScriptEvent(response.data.fixed_height_downsampled_url);
+        });
+
+    };
+
+    function createNewThoughtBubble() {
+        giphyMoodThoughtBubble = Entities.addEntity({
+            "clientOnly": 1,
+            "dimensions": {
+                "x": 0.11244157701730728,
+                "y": 1.1516592502593994,
+                "z": 1.2351282835006714
+            },
+            "id": "{9d900dff-901b-4b3f-b2f2-0655dbd0ec95}",
+            "modelURL": Script.resolvePath('./resources/ThoughtBubble.fbx'),
+            "name": "ThoughtBubble-Giphy",
+            "parentID": MyAvatar.sessionUUID,
+            "parentJointIndex": MyAvatar.getJointIndex("Head"),
+            "position" : MyAvatar.getJointPosition("HeadTop_End"),
+            "localPosition": {
+                "x": 0.807220458984375,
+                "y": 0.8286701202392578,
+                "z": -0.188949584960
+            },
+            "queryAACube": {
+                "scale": 5.248152732849121,
+                "x": -1.4032782316207886,
+                "y": -11.526137351989746,
+                "z": -2.0149528980255127
+            },
+            "localRotation": {
+                "w": -0.6794636845588684,
+                "x": 0.11407923698425293,
+                "y": -0.7147857546806335,
+                "z": 0.11998607218265533
+            },
+            "type": "Model", 
+            "visible" : false
+        }, true);
+    }
+
+    function createNewWebEntity() {
+        giphyMoodWebEntityID = Entities.addEntity({
+            "clientOnly": 1,
+            "collidesWith": "",
+            "collisionMask": 0,
+            "dimensions": {
+                "x": 0.9472649097442627,
+                "y": 0.5328364968299866,
+                "z": 0.009999999776482582
+            },
+            "id": "{a5a315aa-8169-4767-b66f-ffb27d25a54a}",
+            "name" : "WebEntity-Giphy",
+            "parentID": MyAvatar.sessionUUID, 
+            "parentJointIndex" : MyAvatar.getJointIndex("Head"),
+            "position" : MyAvatar.getJointPosition("HeadTop_End"),
+            "localPosition": {
+                "x":0.8114402294158936,
+                "y":0.9220905303955078,
+                "z":-0.10172748565673828
+            },
+            "queryAACube": {
+                "scale": 3.2606639862060547,
+                "x": -0.30625295639038086,
+                "y": -10.532992362976074,
+                "z": -1.059478998184204
+            },
+            "localRotation": {
+                "x":-0.004176754504442215,
+                "y":-0.9858834147453308,
+                "z":0.1655092090368271,
+                "w":0.024976519867777824
+            },
+            "sourceUrl": "http://giphy.com",
+            "type": "Web",
+            "userData": "{\"grabbableKey\":{\"grabbable\":false}}",
+            "visible" : false
+        }, true);
+    }
```

Comment:
I think you could remove some of the extra attributes like ID and queryAACube, but not required